### PR TITLE
Graphql relation between students and subjects

### PR DESCRIPTION
**Pull request by ijas-ws**
### Ticket Link

---

### Related Links

---

### Description

---

### Steps to Reproduce / Test

---

---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

---

### CODE REVIEW

File: server/gql/models/students/index.js [hunk 1]
```diff
@@ -32,6 +36,17 @@ const StudentConnection = createConnection({
   nodeType: Student,
   before: (findOptions, args, context) => {
     findOptions.include = findOptions.include || [];
+
+    if (context?.subject?.id) {
+      console.log('running or not');
```
Comment (alichherawalla) [inline server/gql/models/students/index.js]: lol, let's remove 😂

File: server/gql/models/tests/students/query.test.js [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
-  it('should request for students', async () => {
+  it('should request for subjects related to the students', async () => {
     const dbClient = mockDBClient();
     resetAndMockDB(null, {}, dbClient);
 
+    jest.spyOn(dbClient.models.subjects, 'findAll').mockImplementation(() => [subjectsTable[0]]);
+
     await getResponse(studentOne).then(response => {
       expect(get(response, 'body.data.student')).toBeTruthy();
+
```
Comment (alichherawalla) [inline server/gql/models/tests/students/query.test.js:30]: let's test against the shapee of the object

File: server/gql/models/students/index.js (hunk 1 above)
Comment (ijas-ws) [reply server/gql/models/students/index.js]: 😎😂

File: server/gql/models/tests/students/query.test.js (hunk 2 above)
Comment (ijas-ws) [reply server/gql/models/tests/students/query.test.js:30]: I did this test case in the next PR, is it okay?
That test includes limit and offset scenario also.

Comment (alichherawalla) [reply server/gql/models/tests/students/query.test.js:30]: fair

**Review by alichherawalla — APPROVED** [review]

**Pull request merged**